src/twisted/names/dns.py

```diff
@@ -2005,13 +2088,89 @@ class Record_SPF(Record_TXT):
     @ivar data: Freeform text which makes up this record.
 
     @type ttl: L{int}
-    @ivar ttl: The maximum number of seconds which this record should be cached.
+    @ivar ttl: The maximum number of seconds
+               which this record should be cached.
     """
     TYPE = SPF
     fancybasename = 'SPF'
 
 
 
+@implementer(IEncodable, IRecord)
+class Record_TSIG(tputil.FancyEqMixin, tputil.FancyStrMixin):
+    """
+    A transaction signature, encapsulated in a RR, as described
+    in U{RFC 2845 <https://tools.ietf.org/html/rfc2845>}.
+
+    @type algorithm: L{Name}
+    @ivar algorithm: The name of the signature or MAC algorithm.
+
+    @type time: L{int}
+    @ivar time: Signing time, as seconds from the POSIX epoch.
+
+    @type fudge: L{int}
+    @ivar fudge: Allowable time skew, in seconds.
+
+    @type MAC: L{bytes}
+    @ivar MAC: The message digest or signature.
+
+    @type originalID: L{int}
+    @ivar originalID: A message ID.
+
+    @type otherData: L{bytes}
+    @ivar otherData: Other data carried in exceptional cases.
+
+    """
+    fancybasename = "TSIG"
+    compareAttributes = ('algorithm', 'time', 'fudge',
+                         'MAC', 'originalID', 'error', 'otherData',
+                         'ttl')
+    showAttributes = ['algorithm', 'time', 'MAC', 'error', 'otherData']
+
+    TYPE = TSIG
+
+    def __init__(self, algorithm=None, time=None,
+                 fudge=5, MAC=None, originalID=0,
+                 error=OK, otherData=b'', ttl=0):
```

Comment:
since TTL is always 0 in RFC, do we want it to expose it ?
We can add it later, if needed... but until then https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it

Reply:
The `ttl` field doesn't even belong on this class, since TTL is a property of the record header, not of the record data. But t.n.dns expects one. 

(Actually, it looks like it would be OK to have a `ttl` keyword argument but ignore it and have no `ttl` attribute. I think that would be worse; I'd rather have all the record classes be weird in the same way, than have them be inconsistent.)

Reply:
It seems to me that if this record type has special rules around its TTL, that should at least be documented in its `@ivar`.  (Which, btw, this doesn't have; it should have one.)